apps/omg_watcher_info/lib/omg_watcher_info/db/block.ex

```diff
@@ -92,6 +92,67 @@ defmodule OMG.WatcherInfo.DB.Block do
     |> Paginator.set_data(paginator)
   end
 
+  @spec query_count :: Ecto.Query.t()
+  defp query_count do
+    from(block in __MODULE__, select: count())
+  end
+
+  @spec query_timestamp_between(Ecto.Query.t(), non_neg_integer(), non_neg_integer()) :: Ecto.Query.t()
+  def query_timestamp_between(query, start_datetime, end_datetime) do
+    from(block in query,
+      where:
+        block.timestamp >= ^start_datetime and
+          block.timestamp <= ^end_datetime
+    )
+  end
+
+  @doc """
+  Returns the total number of blocks in between given timestamps
+  """
+  @spec count_all_between_timestamps(non_neg_integer(), non_neg_integer()) :: non_neg_integer()
+  def count_all_between_timestamps(start_datetime, end_datetime) do
+    query_count()
+    |> query_timestamp_between(start_datetime, end_datetime)
+    |> DB.Repo.all()
+    |> Enum.at(0)
```

Comment:
Instead of:
```
  |> DB.Repo.all()
  |> Enum.at(0)
```
how about 
```
DB.Repo.one!()
```
I think this represents what you want more.

Reply:
Thanks Jacob. Done 7fbf75b.